rtps-r52/tests/mailbox.c

```diff
@@ -26,3 +32,58 @@ int test_mbox_lsio_loopback(void)
     test_end("test_mbox_lsio_loopback", rc);
     return rc;
 }
+
+int test_mbox_rtps_hpps(unsigned mbox_in, unsigned mbox_out){
+
+    enum hpsc_mbox_test_rc rc = HPSC_MBOX_TEST_SUCCESS;
+
+    test_begin("test_mbox_rtps_hpps");   
+
+    if( mbox_in >= HPSC_MBOX_CHANNELS || mbox_out >= HPSC_MBOX_CHANNELS ){
+        rc = HPSC_MBOX_TEST_CHAN_CLAIM;
+        test_end("test_mbox_rtps_hpps",rc);
+        return(rc);    
+    }
+
+    struct hpsc_mbox *mbox_hpps = dev_get_mbox(DEV_ID_MBOX_HPPS_RTPS); 
+
+    assert(mbox_hpps);
+
+    const char* name = "TEST_LINK";
+    struct link *test_link = link_mbox_connect(name, mbox_hpps,mbox_in ,mbox_out , 0, 1); 
+    assert(link);
+
+    int bytes; //bytes read 
+
+    char write_buf [HPSC_MSG_SIZE];
+    char read_buf [HPSC_MSG_SIZE];
+    char payload [HPSC_MSG_PAYLOAD_SIZE];
+
+    memset(&payload, 0, sizeof(payload));
+    hpsc_msg_ping(write_buf, HPSC_MSG_SIZE, payload, sizeof(payload));
```

Comment:
On closer inspection, this does not appear correct to me - please double check though.  The `&payload` param to `memset` makes a double pointer, which is probably not what you want.  I think it should just `payload`, which is already a pointer to the array.  You can avoid the `memset` call altogether by initializing `payload` with a value, or to 0, e.g., `char payload[HPSC_MSG_PAYLOAD_SIZE] = { 0 };` or using the macro `HPSC_MSG_DEFINE` in `hpsc-msg.h`, e.g.,  `HPSC_MSG_DEFINE(payload);`.

If you're not going to put any actual content in the payload, you can drop it altogether and just pass `NULL` and size `0` for the respective params to `hpsc_msg_ping`.